### ISSUE TITLE
Export to Zip file

### ISSUE THREAD

**Issue by sammachin** [flowforge/flowforge#273]
### Epic

#268

### Description

**As a:** Owner

**I want to:** Export components of a project to a ZIP file

**So that:** I can store a backup or move to another flowforge system or node-red instance

**Comment by sammachin** [comment]
We should look at the Node-RED project structure and see if we can be compatible with that, possibly via a small transformation script

**Comment by sammachin** [comment]
We need to consider corresponding stories for what the user would do with the Zip file, is there an 'import from zip file' so that a user can restore from their own external backup, Is there a way to run a project locally from a Zip file with local Node-RED? Export without Import doesn't really provide any value.

**Comment by hardillb** [comment]
If the export was done right....

The zip could contain a userDir, so would contain

- settings.js file with all the settings from the template included
- package.json with any extra nodes from the template/stack
- flow.json and flow_creds.json

When unzipped could Node-RED could be pointed at the directory (would need `npm install` running to import the nodes)

Just a thought

**Comment by sammachin** [comment]
Yes this is the sort of thing I was thinking along the lines of, while this seems like a good story, exporting to a self managed Node-RED doesn't create a lot of value for FlowForge it just encourages you to take projects off the platform.

**Comment by knolleary** [comment]
@sammachin you are right to say we need to consider the import side of this as well.

Being able to import back into FlowForge and being able to run it in a local Node-RED instance are two quite different use cases in terms of what the zip can/should contain.

For importing it back into FlowForge (either as migration to another instance, or to restore a manual backup), we cannot take the raw settings.js file - we need the Project Settings JSON. And we have to make sure it isn't a way to import configurations that the platform disallows based on the available templates.

For exporting it as a self-contained standalone Node-RED project, as Ben says, that is much more understood in terms of required files contents.

Are the three uses cases - backing up, move to another FF and move to a standalone NR instance - actually stories in their own right that each represents a different piece of customer value and should be split out? I wonder if the Zip file nature of this story is more an implementation detail?

Otherwise, to deliver this story as described, we need to handle all of those scenarios in one go.
